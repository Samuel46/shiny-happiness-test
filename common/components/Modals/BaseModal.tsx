import React, { FC } from 'react';

import styles from '../../../styles/Modals.module.css';
import { useModal } from '../../hooks/useBaseModal';

type BaseModalProps = {
  isOpen: boolean;
  onDismiss?: () => void;
};

const BaseModal: FC<BaseModalProps> = (props) => {
  useModal({ isOpen: props.isOpen });

  return props.isOpen ? (
    <div className={styles.baseModalWrapper}>
      <div className={styles.baseModalContainer}>
        {!!props.onDismiss && (
          <div className={styles.dismissModalButtonContainer}>
            <button
              type="button"
              onClick={props.onDismiss}
              className={styles.dismissModalButton}
            >
              Close
            </button>
          </div>
        )}
        {props.children}
      </div>
    </div>
  ) : null;
};

export { BaseModal };
