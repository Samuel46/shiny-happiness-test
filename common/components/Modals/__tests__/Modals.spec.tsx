import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import { BaseModal } from '../BaseModal';

it('should render', () => {
  const { asFragment } = render(<BaseModal isOpen>Some modal</BaseModal>);

  expect(asFragment()).toMatchSnapshot();
});

it('should call the dismiss modal handler on click', () => {
  const onClick = jest.fn();

  render(
    <BaseModal isOpen onDismiss={onClick}>
      A Modal
    </BaseModal>,
  );

  fireEvent.click(screen.getByRole('button'));

  expect(onClick).toHaveBeenCalled();
});
