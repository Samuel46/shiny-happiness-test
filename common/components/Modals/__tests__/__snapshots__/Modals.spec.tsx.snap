// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<DocumentFragment>
  <div
    class="baseModalWrapper"
  >
    <div
      class="baseModalContainer"
    >
      Some modal
    </div>
  </div>
</DocumentFragment>
`;
